src/pages/api/profile/user-profile/index.ts

```diff
@@ -0,0 +1,116 @@
+
+import { Handler } from "@src/@types/handler";
+import { UserCookie } from "@src/@types/user-cookie";
+import { configVariables } from "@src/configs/config-variables";
+import { logger } from "@src/configs/logger";
+import { initProfileClient } from "@src/configs/profile-client";
+import getTracer from "@src/configs/tracer";
+import { ERRORS_DATA, webErrorFactory } from "@src/errors/web-errors";
+import { authMiddleware } from "@src/middlewares/auth-middleware";
+import { sentryMiddleware } from "@src/middlewares/sentry-middleware";
+import { getProfileAccessToken } from "@src/utils/get-profile-access-token";
+
+const patchHandler: Handler = async (request, response) => {
+  const webErrors = {
+    invalidProfileToken: ERRORS_DATA.INVALID_PROFILE_TOKEN,
+    invalidScopes: ERRORS_DATA.INVALID_SCOPES,
+    userProfileNotFound: ERRORS_DATA.USER_PROFILE_NOT_FOUND,
+    invalidUserProfilePayload: ERRORS_DATA.INVALID_USER_PROFILE_PAYLOAD,
+    unreachable: ERRORS_DATA.UNREACHABLE,
+  };
+
+  return getTracer().span(
+    "/api/profile/user-profile patchHandler",
+    async (span) => {
+      const userCookie = await getServerSideCookies<UserCookie>(request, {
+        cookieName: "user-cookie",
+        isCookieSealed: true,
+        cookieSalt: configVariables.cookieSalt,
+      });
+
+      if (!userCookie) {
+        response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+        response.setHeader("location", "/");
+        response.end();
+        return;
+      }
+
+      const { userProfilePayload } = request.body;
+
+      const profileAccessToken = await getProfileAccessToken(
+        userCookie.access_token,
+      );
+      span.setDisclosedAttribute(
+        "is Connect.Profile access token available",
+        true,
+      );
+
+      const profileClient = initProfileClient(profileAccessToken);
+
+      const { data: updatedUserProfile } = await profileClient
+        .patchProfile(userProfilePayload)
+        .catch((error) => {
+          span.setDisclosedAttribute(
+            "is Connect.Profile UserProfile updated",
+            false,
+          );
+
+          if (error.response.status === HttpStatus.UNAUTHORIZED) {
+            throw webErrorFactory(webErrors.invalidProfileToken);
+          }
+
+          if (error.response.status === HttpStatus.FORBIDDEN) {
+            throw webErrorFactory(webErrors.invalidScopes);
+          }
+
+          if (error.response.status === HttpStatus.NOT_FOUND) {
+            throw webErrorFactory(webErrors.userProfileNotFound);
+          }
+
+          if (error.response.status === HttpStatus.UNPROCESSABLE_ENTITY) {
+            throw webErrorFactory(webErrors.invalidUserProfilePayload);
+          }
+
+          throw webErrorFactory(webErrors.unreachable);
+        });
+      span.setDisclosedAttribute(
+        "is Connect.Profile UserProfile updated",
+        true,
+      );
+
+      response.statusCode = HttpStatus.OK;
+      response.json({ updatedUserProfile });
```

Comment:
Do we want the api to respond with:
```
{
  updatedUserProfile: {
    name: ...,
    otherFields,
  }
}
```
or directly the resource:
```
{
  name: ...,
  otherFields,
}
```

I think we should use the second version and send the profile rather than an object with a `updatedUserProfile` key, so here's my suggestion:
```suggestion
      response.json(updatedUserProfile);
```

Reply:
Yeah, that's what I was proposing on Pauline's PR. Just didn't put any thoughts on this haha. I'll make the changes.

Reply:
Not sure if I should do this now, as it would make this tedious, as each values can be undefined:
```ts
            if ("updatedUserProfile" in parsedResponse) {
              router && router.push("/account/profile");
              return;
            }
            ```

Reply:
Answer here: https://github.com/fewlinesco/connect-account/pull/541#discussion_r626555100